teamcity: failed test: TestLogic

The following tests appear to have failed on master (test): TestLogic/fakedist: TestLogic/fakedist/crdb_internal, TestLogic/fakedist, TestLogic/fakedist: TestLogic/fakedist/crdb_internal: TestLogic/fakedist/crdb_internal/max_retry_counter, TestLogic

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestLogic).

[#1327655](https://teamcity.cockroachdb.com/viewLog.html?buildId=1327655):


```
TestLogic/fakedist: TestLogic/fakedist/crdb_internal
--- FAIL: test/TestLogic: TestLogic/fakedist: TestLogic/fakedist/crdb_internal (2.170s)

------- Stdout: -------
=== PAUSE TestLogic/fakedist/crdb_internal



TestLogic
--- FAIL: test/TestLogic (428.390s)

------- Stdout: -------
    test_log_scope.go:79: test logs captured to: /tmp/logTestLogic244295760
    test_log_scope.go:60: use -show-logs to present logs inline



TestLogic/fakedist: TestLogic/fakedist/crdb_internal: TestLogic/fakedist/crdb_internal/max_retry_counter
--- FAIL: test/TestLogic: TestLogic/fakedist: TestLogic/fakedist/crdb_internal: TestLogic/fakedist/crdb_internal/max_retry_counter (0.020s)

------- Stdout: -------
--- done: testdata/logic_test/crdb_internal with config fakedist: 78 tests, 1 failures
=== CONT  TestLogic/fakedist/collatedstring_uniqueindex2
=== CONT  TestLogic/fakedist/inet
                logic.go:2336: 
                     
                    testdata/logic_test/crdb_internal:434: SELECT key, (max_retries > 1), flags LIKE '!%' AS f
                      FROM crdb_internal.node_statement_statistics
                     WHERE application_name = 'test_max_retry'
                    ORDER BY key, f
                    expected:
                        SELECT crdb_internal.force_retry(_)  true  false
                        SELECT crdb_internal.force_retry(_)  true  true
                        SET application_name = DEFAULT       false false
                    but found (query options: "") :
                        SELECT crdb_internal.force_retry(_)  false  false
                        SELECT crdb_internal.force_retry(_)  false  true
                        SET application_name = DEFAULT       false  false
                        
                    



TestLogic/fakedist
--- FAIL: test/TestLogic: TestLogic/fakedist (0.850s)





```

Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/logictest/testdata/logic_test/crdb_internal
+++ b/pkg/sql/logictest/testdata/logic_test/crdb_internal
@@ -453,13 +453,10 @@ SET application_name = 'test_max_retry'
 
 # Make the statement retry, to ensure max_retries increases to
 # become different from 0.
-# The delay is chosen identically to the value in the `txn`
-# logic tests introduced in #16719. It is long enough
-# to ensure max_retries increases, even under stress,
-# but short enough to be negligible compared to the
-# overall test suite.
 statement OK
-SELECT crdb_internal.force_retry('50ms'::INTERVAL)
+CREATE SEQUENCE s;
+  SELECT IF(nextval('s')<3, crdb_internal.force_retry('1h'::INTERVAL), 0);
+  DROP SEQUENCE s
 
 statement OK
 RESET application_name
@@ -476,15 +473,17 @@ RESET application_name
 # reset to 0 between statements - a naive implementation could make
 # the counter increase forever, even between statements.
 #
-query TBB
-SELECT key, (max_retries > 0), flags LIKE '!%' AS f
+query TIB
+SELECT key, max_retries, flags LIKE '!%' AS f
   FROM crdb_internal.node_statement_statistics
  WHERE application_name = 'test_max_retry'
 ORDER BY key, f
 ----
-SELECT crdb_internal.force_retry(_)  true  false
-SELECT crdb_internal.force_retry(_)  true  true
-SET application_name = DEFAULT       false false
+CREATE SEQUENCE s                                           0  false
+DROP SEQUENCE s                                             0  false
+SELECT IF(nextval(_) < _, crdb_internal.force_retry(_), _)  2  false
+SELECT IF(nextval(_) < _, crdb_internal.force_retry(_), _)  1  true
+SET application_name = DEFAULT                              0  false
 
 
 # Testing split_enforced_until when truncating and dropping.
@@ -523,12 +522,12 @@ query TT colnames
 SELECT start_pretty, end_pretty FROM crdb_internal.ranges WHERE split_enforced_until IS NOT NULL
 ----
 start_pretty   end_pretty
-/Table/57/1/1  /Table/57/1/2
-/Table/57/1/2  /Table/57/1/3
-/Table/57/1/3  /Table/57/2/1
-/Table/57/2/1  /Table/57/2/2
-/Table/57/2/2  /Table/57/2/3
-/Table/57/2/3  /Max
+/Table/58/1/1  /Table/58/1/2
+/Table/58/1/2  /Table/58/1/3
+/Table/58/1/3  /Table/58/2/1
+/Table/58/2/1  /Table/58/2/2
+/Table/58/2/2  /Table/58/2/3
+/Table/58/2/3  /Max
 
 statement ok
 DROP TABLE foo
@@ -551,12 +550,12 @@ query TT colnames
 SELECT start_pretty, end_pretty FROM crdb_internal.ranges WHERE split_enforced_until IS NOT NULL
 ----
 start_pretty   end_pretty
-/Table/58/1/1  /Table/58/1/2
-/Table/58/1/2  /Table/58/1/3
-/Table/58/1/3  /Table/58/2/1
-/Table/58/2/1  /Table/58/2/2
-/Table/58/2/2  /Table/58/2/3
-/Table/58/2/3  /Max
+/Table/59/1/1  /Table/59/1/2
+/Table/59/1/2  /Table/59/1/3
+/Table/59/1/3  /Table/59/2/1
+/Table/59/2/1  /Table/59/2/2
+/Table/59/2/2  /Table/59/2/3
+/Table/59/2/3  /Max
 
 statement ok
 DROP INDEX foo@idx
@@ -567,6 +566,6 @@ query T colnames
 SELECT start_pretty FROM crdb_internal.ranges WHERE split_enforced_until IS NOT NULL
 ----
 start_pretty
-/Table/58/1/1
-/Table/58/1/2
-/Table/58/1/3
+/Table/59/1/1
+/Table/59/1/2
+/Table/59/1/3

--- a/pkg/sql/logictest/testdata/logic_test/manual_retry
+++ b/pkg/sql/logictest/testdata/logic_test/manual_retry
@@ -2,17 +2,31 @@
 
 subtest automatic_retry
 
+statement ok
+CREATE SEQUENCE s
+
 # On an implicit transaction, we retry automatically and the function
 # eventually returns a result.
 query I
-SELECT crdb_internal.force_retry('50ms':::INTERVAL)
+SELECT IF(nextval('s')<3, crdb_internal.force_retry('1h':::INTERVAL), 0)
 ----
 0
+
+# Demonstrate that the txn was indeed retried.
+query I
+SELECT currval('s')
+----
+3
+
+statement ok
+DROP SEQUENCE s
 
 subtest automatic_retry
 
 statement ok
-BEGIN TRANSACTION; SAVEPOINT cockroach_restart
+CREATE SEQUENCE s;
+  BEGIN TRANSACTION;
+  SAVEPOINT cockroach_restart
 
 # The SELECT 1 is necessary to take the session out of the AutoRetry state,
 # otherwise the statement below would be retries automatically.
@@ -20,31 +34,45 @@ statement ok
 SELECT 1
 
 query error restart transaction: crdb_internal.force_retry\(\): TransactionRetryWithProtoRefreshError: forced by crdb_internal.force_retry\(\)
-SELECT crdb_internal.force_retry('500ms':::INTERVAL)
+SELECT crdb_internal.force_retry('1h':::INTERVAL)
 
 statement ok
 ROLLBACK TO SAVEPOINT cockroach_restart
 
-# wait until the transaction is at least 1 second
-sleep 1s
+query I
+SELECT IF(nextval('s')<3, crdb_internal.force_retry('1h':::INTERVAL), 0)
+----
+0
+
+# Demonstrate that the txn was indeed retried.
+query I
+SELECT currval('s')
+----
+3
+
+statement ok
+COMMIT
+
+subtest savepoint_name
+
+statement ok
+BEGIN
 
 # Ensure that ident case rules are used.
 statement error pq: unimplemented: SAVEPOINT not supported except for cockroach_restart
 SAVEPOINT "COCKROACH_RESTART"
 
+statement ok
+ROLLBACK; BEGIN
+
 # Ensure that ident case rules are used.
 statement ok
 SAVEPOINT COCKROACH_RESTART
 
-query I
-SELECT crdb_internal.force_retry('500ms':::INTERVAL)
-----
-0
-
 statement ok
-COMMIT
+ROLLBACK
 
-subtest schema_chage_with_rollback
+subtest schema_change_with_rollback
 
 # Test that creating a table repeatedly across restarts doesn't leave dangling
 # rows behind (the rows are  associated with the correct descriptor).

--- a/pkg/sql/logictest/testdata/logic_test/txn
+++ b/pkg/sql/logictest/testdata/logic_test/txn
@@ -558,7 +558,7 @@ statement ok
 BEGIN TRANSACTION; SAVEPOINT cockroach_restart; SELECT 1
 
 query error pgcode 40001 restart transaction: crdb_internal.force_retry\(\): TransactionRetryWithProtoRefreshError: forced by crdb_internal.force_retry\(\)
-SELECT crdb_internal.force_retry('1s':::INTERVAL)
+SELECT crdb_internal.force_retry('1h':::INTERVAL)
 
 query T
 SHOW TRANSACTION STATUS
@@ -578,31 +578,60 @@ COMMIT
 
 
 # Automatic retries for the first batch.
+# We use a sequence to avoid busy-looping the test.
 statement ok
-BEGIN TRANSACTION; SELECT crdb_internal.force_retry('100ms':::INTERVAL)
+CREATE SEQUENCE s;
+  BEGIN TRANSACTION;
+  SELECT IF(nextval('s')<3, crdb_internal.force_retry('1h':::INTERVAL), 0)
+
+# Demonstrate that the txn was indeed retried.
+query I
+SELECT currval('s')
+----
+3
 
 statement ok
-ROLLBACK
+ROLLBACK;
+  DROP SEQUENCE s
 
 # Automatic retries for the first batch even when that first batch comes after
 # the BEGIN.
 statement ok
-BEGIN TRANSACTION;
+CREATE SEQUENCE s;
+  BEGIN TRANSACTION;
 
 statement ok
-SELECT 1; SELECT crdb_internal.force_retry('100ms':::INTERVAL)
+SELECT 1;
+  SELECT IF(nextval('s')<3, crdb_internal.force_retry('1h':::INTERVAL), 0)
+
+# Demonstrate that the txn was indeed retried.
+query I
+SELECT currval('s')
+----
+3
 
 statement ok
-ROLLBACK
+ROLLBACK;
+  DROP SEQUENCE s
 
 # Automatic retries for the first batch even when that first batch comes after
 # the BEGIN and the BEGIN also has special statements that don't move the txn
 # state out of the "AutoRetry" state.
 statement ok
-BEGIN TRANSACTION; SAVEPOINT cockroach_restart; SET TRANSACTION PRIORITY HIGH; SET TRANSACTION ISOLATION LEVEL SNAPSHOT;
+CREATE SEQUENCE s;
+  BEGIN TRANSACTION;
+  SAVEPOINT cockroach_restart;
+  SET TRANSACTION PRIORITY HIGH;
+  SET TRANSACTION ISOLATION LEVEL SNAPSHOT;
 
 statement ok
-SELECT crdb_internal.force_retry('100ms':::INTERVAL)
+SELECT IF(nextval('s')<3, crdb_internal.force_retry('1h':::INTERVAL), 0)
+
+# Demonstrate that the txn was indeed retried.
+query I
+SELECT currval('s')
+----
+3
 
 query T
 SHOW TRANSACTION ISOLATION LEVEL
@@ -615,25 +644,37 @@ SHOW TRANSACTION PRIORITY
 high
 
 statement ok
-ROLLBACK
+ROLLBACK;
+  DROP SEQUENCE s
 
 # Like above, but the SAVEPOINT is its own batch.
 statement ok
-BEGIN TRANSACTION
+CREATE SEQUENCE s;
+  BEGIN TRANSACTION
 
 statement ok
 SAVEPOINT cockroach_restart;
 
 statement ok
-SELECT crdb_internal.force_retry('100ms':::INTERVAL)
+SELECT IF(nextval('s')<3, crdb_internal.force_retry('1h':::INTERVAL), 0)
+
+# Demonstrate that the txn was indeed retried.
+query I
+SELECT currval('s')
+----
+3
 
 statement ok
-ROLLBACK
+ROLLBACK;
+  DROP SEQUENCE s
 
 
 # Automatic retries for the first batch after an explicit restart.
 statement ok
-BEGIN TRANSACTION; SAVEPOINT cockroach_restart; SELECT 1;
+CREATE SEQUENCE s;
+  BEGIN TRANSACTION;
+  SAVEPOINT cockroach_restart;
+  SELECT 1;
 
 query error pgcode 40001 restart transaction: crdb_internal.force_retry\(\): TransactionRetryWithProtoRefreshError: forced by crdb_internal.force_retry\(\)
 SELECT crdb_internal.force_retry('1h':::INTERVAL)
@@ -643,18 +684,27 @@ ROLLBACK TO SAVEPOINT COCKROACH_RESTART;
 
 # This is the automatic retry we care about.
 statement ok
-SELECT crdb_internal.force_retry('50ms':::INTERVAL)
+SELECT IF(nextval('s')<3, crdb_internal.force_retry('1h':::INTERVAL), 0)
+
+# Demonstrate that the txn was indeed retried.
+query I
+SELECT currval('s')
+----
+3
 
 statement ok
-ROLLBACK
+ROLLBACK;
+  DROP SEQUENCE s
 
 
 # Wrong savepoint name moves the txn state from RestartWait to Aborted.
 statement ok
-BEGIN TRANSACTION; SAVEPOINT cockroach_restart; SELECT 1;
+BEGIN TRANSACTION;
+  SAVEPOINT cockroach_restart;
+  SELECT 1
 
 query error pgcode 40001 restart transaction: crdb_internal.force_retry\(\): TransactionRetryWithProtoRefreshError: forced by crdb_internal.force_retry\(\)
-SELECT crdb_internal.force_retry('1s':::INTERVAL)
+SELECT crdb_internal.force_retry('1h':::INTERVAL)
 
 query T
 SHOW TRANSACTION STATUS
@@ -974,7 +1024,7 @@ statement ok
 BEGIN; SELECT 1
 
 query error pgcode 40001 restart transaction: crdb_internal.force_retry\(\): TransactionRetryWithProtoRefreshError: forced by crdb_internal.force_retry\(\)
-SELECT crdb_internal.force_retry('1s':::INTERVAL)
+SELECT crdb_internal.force_retry('1h':::INTERVAL)
 
 statement ok
 ROLLBACK

